src/browser/pages/vscode.ts

```diff
@@ -54,39 +52,35 @@ export function getNlsConfiguration(document: Document) {
   return JSON.parse(nlsConfig) as NlsConfiguration
 }
 
-try {
-  const nlsConfig = getNlsConfiguration(document)
-  if (nlsConfig._resolvedLanguagePackCoreLocation) {
-    const bundles = Object.create(null)
-    nlsConfig.loadBundle = (bundle: any, _language: any, cb: any): void => {
-      const result = bundles[bundle]
-      if (result) {
-        return cb(undefined, result)
-      }
-      // FIXME: Only works if path separators are /.
-      const path = nlsConfig._resolvedLanguagePackCoreLocation + "/" + bundle.replace(/\//g, "!") + ".nls.json"
-      fetch(`${options.base}/vscode/resource/?path=${encodeURIComponent(path)}`)
-        .then((response) => response.json())
-        .then((json) => {
-          bundles[bundle] = json
-          cb(undefined, json)
-        })
-        .catch(cb)
-    }
+type GetLoaderParams = {
+  origin: string
+  nlsConfig: NlsConfiguration
+  options: Options
+}
+
+type Loader = {
+  baseUrl: string
+  recordStats: boolean
+  paths: {
+    [key: string]: string
   }
-  ;(self.require as any) = {
+  "vs/nls": NlsConfiguration
+}
+
+/**
+ * A helper function to get the require loader
+ *
+ * This used by VSCode/code-server
+ * to load files.
+ *
+ * We extracted the logic into a function so that
+ * it's easier to test.
+ **/
+export function getLoader({ origin, nlsConfig, options }: GetLoaderParams) {
+  const loader: Loader = {
     // Without the full URL VS Code will try to load file://.
-    baseUrl: `${window.location.origin}${options.csStaticBase}/lib/vscode/out`,
+    baseUrl: `${origin}${options.csStaticBase}/lib/vscode/out`,
     recordStats: true,
-    // TODO: There don't appear to be any types for trustedTypes yet.
-    trustedTypesPolicy: (window as any).trustedTypes?.createPolicy("amdLoader", {
-      createScriptURL(value: string): string {
-        if (value.startsWith(window.location.origin)) {
-          return value
-        }
-        throw new Error(`Invalid script url: ${value}`)
-      },
-    }),
```

Comment:
`trustedTypesPolicy` was removed?

Reply:
Um...that's weird. I don't remember doing that? Maybe a bad rebase on my part? most likely. I'lll add in!